src/EditDistPeriodicityDetector.cpp

```diff
@@ -45,70 +46,95 @@ namespace geopm
 {
     EditDistPeriodicityDetector::EditDistPeriodicityDetector(int history_buffer_size)
         : m_history_buffer(history_buffer_size)
+        , m_history_buffer_size(history_buffer_size)
         , m_period(-1)
         , m_score(-1)
+        , m_record_count(0)
+        , m_DP(history_buffer_size * history_buffer_size * history_buffer_size)
     {
-
+        //m_DP = geopm::make_unique<std::vector<uint32_t> >(history_buffer_size * history_buffer_size * history_buffer_size);
+        m_myinf = 2*history_buffer_size;
     }
 
     void EditDistPeriodicityDetector::update(const record_s &record)
     {
         if (record.event == EVENT_REGION_ENTRY) {
             m_history_buffer.insert(record.signal);
+            m_record_count ++;
             calc_period();
         }
     }
 
+    void EditDistPeriodicityDetector::Dset(int ii, int jj, int mm, uint32_t val) {
+        m_DP[((ii % m_history_buffer_size) * m_history_buffer_size + (jj % m_history_buffer_size)) * m_history_buffer_size + (mm % m_history_buffer_size)] = val;
```

Comment:
Add assertion to check boundaries. Get the name of the ASSERTION function from Diana.

Split up the line to make more readable.